std/testing/asserts_test.ts

```diff
@@ -124,8 +124,8 @@ Deno.test("testingEqual", function (): void {
 });
 
 Deno.test("testingNotEquals", function (): void {
-  const a = { foo: "bar" };
-  const b = { bar: "foo" };
+  const a = { foo: "bar" } as object;
+  const b = { bar: "foo" } as object;
```

Comment:
Seems like a net negative to me if you need to add `as object` before asserting.

Reply:
Thanks for the review. I think it's only necessary for this case (because we want to test that the assert function can really differentiate these two values), but under common usage, users probably will not run into situation where they need to assert the equality of two values that has different types in Typescript. Most of the time if this happens, it's probably that the user is comparing the wrong values.

Reply:
I do understand that in some cases, the user will intend to compare values of different types, for example [this line](https://github.com/denoland/deno/blob/ffedbd79ad90bc88ef8c51d145536f68f50d4fea/std/hash/fnv_test.ts#L45) where the user is trying to compare `string | number[]` to `Uint8Array`. But from my point of view, this kind of assertion is usually rare for higher level code, so it might be worth it to compromise the testing experience for these rare cases in exchange for better overall testing experience.